rllib/core/rl_module/marl_module.py

```diff
@@ -0,0 +1,295 @@
+        """
+        for module_id, module in self._rl_modules.items():
+            module.set_state(state_dict[module_id])
+
+    @override(RLModule)
+    def as_multi_agent(self) -> "MultiAgentRLModule":
+        """Returns a multi-agent wrapper around this module.
+
+        This method is overridden to avoid double wrapping.
+
+        Returns:
+            The instance itself.
+        """
+        return self
+
+    def add_module(
+        self,
+        module_id: ModuleID,
+        module: RLModule,
+        *,
+        dist_config=None,
+        override: bool = False,
+    ) -> None:
+        """Adds a module at run time to the multi-agent module.
+
+        Args:
+            module_id: The module ID to add. If the module ID already exists and
+                override is False, an error is raised. If override is True, the module
+                is replaced.
+            module: The module to add.
+            dist_config: The distributed configuration to use if module needs to be
+                distributed.
+            override: Whether to override the module if it already exists.
+
+        Raises:
+            ValueError: If the module ID already exists and override is False.
+        """
+        if module_id in self._rl_modules and not override:
+            raise ValueError(
+                f"Module ID {module_id} already exists. If your intention is to "
+                "override, set override=True."
+            )
+        if self.is_distributed():
+            module.make_distributed(dist_config)
+        self._rl_modules[module_id] = module
+
+    def remove_module(
+        self, module_id: ModuleID, *, raise_err_if_not_found: bool = True
+    ) -> None:
+        """Removes a module at run time from the multi-agent module.
+
+        Args:
+            module_id: The module ID to remove.
+            raise_err_if_not_found: Whether to raise an error if the module ID is not
+                found.
+        Raises:
+            ValueError: If the module ID does not exist and raise_err_if_not_found is
+                True.
+        """
+        if raise_err_if_not_found:
+            self._check_module_exists(module_id)
+        del self._rl_modules[module_id]
+
+    def __getitem__(self, module_id: ModuleID) -> RLModule:
+        """Returns the module with the given module ID.
+
+        Args:
+            module_id: The module ID to get.
+
+        Returns:
+            The module with the given module ID.
+        """
+        self._check_module_exists(module_id)
+        return self._rl_modules[module_id]
+
+    @OverrideToImplementCustomLogic
+    def _make_modules(self) -> Mapping[ModuleID, RLModule]:
+        modules_info_dict = self.config["modules"]
+        modules = {}
+        for module_id, module_info in modules_info_dict.items():
+
+            if isinstance(module_info, RLModule):
+                modules[module_id] = module_info
+            elif isinstance(module_info, dict):
```

Comment:
Hmm, can we completely avoid these dict configs from here on across all of RLlib?

Instead, create a specific `RLModuleConfig` class that has all the information in it for a RLModule to be built from it:

100% analogous to AlgorithmConfig:
```
module_config = RLModuleConfig(rl_module_class=..., setting_a=..., setting_b=...)
my_rl_module = module_config.build()
```

Reply:
Let's just replace these module configs once we have a good way of specifying serializable configs everywhere. I took a stab at it here, but it's more complicated than this PR when it comes to inheritance.